qa/L0_request_cancellation/grpc_cancellation_test.py

```diff
@@ -50,6 +50,29 @@ def callback(user_data, result, error):
         user_data._completed_requests.put(result)
 
 
+def prepare_inputs_outputs():
+    inputs, outputs = [], []
+    inputs.append(grpcclient.InferInput("INPUT0", [1, 1], "INT32"))
+    outputs.append(grpcclient.InferRequestedOutput("OUTPUT0"))
+    inputs[0].set_data_from_numpy(np.array([[5]], dtype=np.int32))
+    return inputs, outputs
+
+
+def grpc_async_infer_request_with_instant_cancellation(model_name):
+    inputs_, outputs_ = prepare_inputs_outputs()
+    user_data = UserData()
+    with grpcclient.InferenceServerClient(url="localhost:8001") as client:
+        t_end = time.time() + 60
+        while time.time() < t_end:
```

Comment:
How about adding artificial delays in ModelInferHandler Process loop? 
That would ensure that we hit cancellation in START step even with a single request instead of running high inference load and expect some of the requests to hit the case.
Something like these: https://github.com/triton-inference-server/server/blob/2ac5f34b3b6dc07ac4c30f38c5df74fa2547fcb9/src/grpc/infer_handler.h#L1016

Reply:
Can this be done in a follow up PR? I feel like we still have to analyze all edge case that are left and not processed?

Reply:
re-factored